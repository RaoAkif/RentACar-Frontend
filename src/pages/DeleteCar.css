.delete-car-main {
  width: 100%;
  height: 100vh;
  margin: auto 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image:
    linear-gradient(black, black),
    url("/src/assets/images/car-wallpaper.jpg");
  background-blend-mode: saturation;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  box-shadow: inset 0 0 0 2000px rgba(212, 216, 221, 0.9);
}

.d-cars-delete-cards {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.d-car-image-div {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.d-car-image {
  width: 100px;
}

.delete-table {
  width: 90%;
  margin-bottom: 3rem;
  font-size: 1.6rem;
}

.delete-table tr {
  padding: 4rem;
}

.delete-table tr td {
  padding: 1rem;
}

.th-checkbox {
  width: 10px;
}

.td-checkbox {
  padding: 18rem;
}

@media only screen and (min-width: 480px) {
  .delete-table {
    width: 30%;
  }
}
