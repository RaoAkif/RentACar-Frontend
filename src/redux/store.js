import { configureStore } from '@reduxjs/toolkit';
import cakeReducer from './cake/cakeSlice';
import icecreamReducer from './icecream/icecreamSlice';
import userReducer from './user/userSlice';

const store = configureStore({
  reducer: {
    cake: cakeReducer,
    icecream: icecreamReducer,
    user: userReducer,
  },
});

export default store;
