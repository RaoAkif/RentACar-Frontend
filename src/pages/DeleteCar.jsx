import React from 'react';
import DeleteCarCard from '../components/DeleteCarCard';
import './DeleteCar.css';

function DeleteCar() {
  const cars = [
    {
      id: '1', name: 'BMW', model: 'M3', rent: '$100', image: 'https://raw.githubusercontent.com/CodeBitChips/Images/main/cars/bmw-m3.png', desc: 'This is a description of the car',
    },
    {
      id: '2', name: 'Audi', model: 'A4', rent: '$100', image: 'https://raw.githubusercontent.com/CodeBitChips/Images/main/cars/bmw-m3.png', desc: 'This is a description of the car',
    },
    {
      id: '3', name: 'BMW', model: 'M3', rent: '$100', image: 'https://raw.githubusercontent.com/CodeBitChips/Images/main/cars/bmw-m3.png', desc: 'This is a description of the car',
    },
    {
      id: '4', name: 'Audi', model: 'A4', rent: '$100', image: 'https://raw.githubusercontent.com/CodeBitChips/Images/main/cars/bmw-m3.png', desc: 'This is a description of the car',
    },
  ];

  return (
    <div className="delete-car-main">
      <h2>DELETE&nbsp;&nbsp;CAR</h2>
      <form className="d-cars-delete-cards">
        <table className="table table-striped table-bordered delete-table">
          <thead className="table-dark">
            <tr>
              <th className="th-checkbox" scope="col"> </th>
              <th scope="col">Company</th>
              <th scope="col">Model</th>
            </tr>
          </thead>
          <tbody>
            {cars.map((car) => (
              <DeleteCarCard
                key={car.id}
                name={car.name}
                model={car.model}
              />
            ))}
          </tbody>
        </table>
        <button className="delete-btn" type="submit">Delete</button>
      </form>
    </div>
  );
}

export default DeleteCar;
