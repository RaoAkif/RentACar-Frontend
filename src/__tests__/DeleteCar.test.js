import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import { render } from '@testing-library/react';
import DeleteCar from '../pages/DeleteCar';

describe('DeleteCar', () => {  
  it('should render correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <DeleteCar />
        </Provider>,
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
