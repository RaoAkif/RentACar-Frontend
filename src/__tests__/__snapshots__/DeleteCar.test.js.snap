// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteCar should render correctly 1`] = `
<div>
  <h2>
    There are no cars
  </h2>
  ,
</div>
`;
